src/types/user.ts

```diff
@@ -54,6 +55,8 @@ export const User = t.intersection([
 
 export type User = t.TypeOf<typeof User>;
 
+export type UserWithMyPortalToken = User & { myportal_token: MyPortalToken };
```

Comment:
I'd rather keep the name 'User' for the latest version (`UserWithMyPortalToken`) in this specific case and rename
`User` -> `UserWithoutMyPortalToken = Omit<User, "myportal_token">`  otherwise you'll end up with

`UserWithMyportalTokenAndBdpTokenAndFaToken...`

do we need this distinction anyway ?

another way is to have something like:

```ts
// without my_portal_token
RequiredUserTokensV1 = t.interface({ session_token: ..., wallet_token: ...})
UserV1 = t.intersection([UserWithoutTokens,RequiredUserTokensV1]);

// latest user version with all required tokens
RequiredUserTokens = t.interface({ session_token: ..., wallet_token: ..., myportal_token: ...})
User = t.intersection([UserWithoutTokens, RequiredUserTokens])
```

so you can check `UserV1.is()` instead of `user.myportal_token === undefined` and have a mapping method UserV1 -> User

Reply:
fixed in c4e4093